src/scheduler/MpiWorld.cpp

```diff
@@ -193,77 +171,36 @@ void MpiWorld::closeThreadLocalClients()
     functionCallClients.clear();
 }
 
-void MpiWorld::initialiseFromState(const faabric::Message& msg, int worldId)
+void MpiWorld::initialiseFromMsg(const faabric::Message& msg, bool forceLocal)
 {
-    id = worldId;
+    id = msg.mpiworldid();
     user = msg.user();
     function = msg.function();
+    size = msg.mpiworldsize();
 
-    setUpStateKV();
-
-    // Read from state
-    MpiWorldState s{};
-    stateKV->pull();
-    stateKV->get(BYTES(&s));
-    size = s.worldSize;
     threadPool = std::make_shared<faabric::scheduler::MpiAsyncThreadPool>(
       getMpiThreadPoolSize());
-}
 
-void MpiWorld::pushToState()
-{
-    // Write to state
-    MpiWorldState s{
-        .worldSize = this->size,
-    };
-
-    stateKV->set(BYTES(&s));
-    stateKV->pushFull();
-}
-
-void MpiWorld::registerRank(int rank)
-{
-    {
-        faabric::util::FullLock lock(worldMutex);
-        rankHostMap[rank] = thisHost;
+    // Sometimes for testing purposes we may want to initialise a world in the
+    // _same_ host we have created one (note that this would never happen in
+    // reality). If so, we skip the rank-host map broadcasting.
+    if (!forceLocal) {
+        // Block until we receive
+        faabric::MpiHostRankMsg hostRankMsg =
+          faabric::transport::recvMpiHostRankMsg();
+        std::vector<std::string> rankHostVec(hostRankMsg.hosts_size());
+        for (int i = 0; i < rankHostVec.size(); i++) {
+            rankHostVec.at(i) = hostRankMsg.hosts(i);
+        }
```

Comment:
Can we just use the copy constructor here and avoid the loop, i.e. 

```
std::vector<std::string> rankHostVec = hostRankMsg.hosts()
```

Or even better, can't we just pass `hostRankMsg.hosts()` straight into `setHostForRank`?

If the copying is necessary, might be good to add a comment explaining why.